Harvest.Web/Controllers/Api/ExpenseController.cs

```diff
@@ -33,16 +33,23 @@ public ExpenseController(AppDbContext dbContext, IUserService userService, IProj
         [Consumes(MediaTypeNames.Application.Json)]
         public async Task<ActionResult> Create(int projectId, [FromBody] Expense[] expenses)
         {
-            // TODO: validation!
+            var project = await _dbContext.Projects.SingleAsync(p => p.Id == projectId);
+            if (project.Status != Project.Statuses.Active
+                && project.Status != Project.Statuses.AwaitingCloseout
+                && project.Status != Project.Statuses.PendingCloseoutApproval)
+            {
+                return BadRequest($"Expenses cannot be created for project with status of {project.Status}");
+            }
+
             var user = await _userService.GetCurrentUser();
             var allRates = await _dbContext.Rates.Where(a => a.IsActive).ToListAsync();
             foreach (var expense in expenses)
             {
-                expense.CreatedBy     = user;
-                expense.CreatedOn     = DateTime.UtcNow;
-                expense.ProjectId     = projectId;
-                expense.InvoiceId     = null;
-                expense.Account       = allRates.Single(a => a.Id == expense.RateId).Account;
+                expense.CreatedBy = user;
+                expense.CreatedOn = DateTime.UtcNow;
+                expense.ProjectId = projectId;
+                expense.InvoiceId = null;
+                expense.Account = allRates.Single(a => a.Id == expense.RateId).Account;
```

Comment:
What do you have against tab alignment?

Reply:
Nothing. Should I disable autoformat for c# code?